staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -124,6 +124,12 @@ func IsDNS1123Subdomain(value string) []string {
 	if !dns1123SubdomainRegexp.MatchString(value) {
 		errs = append(errs, RegexError(dns1123SubdomainErrorMsg, dns1123SubdomainFmt, "example.com"))
 	}
+	subs := strings.Split(value, ".")
```

Comment:
I will not reject using regex, but seems people don't suggest using regex, see the comments here:https://github.com/kubernetes/kubernetes/pull/43525#discussion_r108003174